### DISCUSSION TITLE
Error when adding a primary key before saving the table

### DISCUSSION THREAD

**Discussion by Sod-Almighty** [dbeaver/dbeaver#14353] [Q&A]
#### System information: 
Windows 10 64-bit
DBeaver 21.2.3.xxxxx

#### Connection specification:
SQLite

#### Describe the problem you're observing:

Adding a primary key before saving (persisting) the table to the database results in an error.

#### Steps to reproduce, if exist:

1. Create a new table
2. Create a new integer column, not null, auto-increment

![Prism  Microsoft Remote Desktop, Today at 1 23 45 pm](https://user-images.githubusercontent.com/4511095/138453020-877ce4e5-5f69-49fe-904a-e2d835e643c6.png)

3. Go to the Keys section and add a primary key on that column

![Prism  Microsoft Remote Desktop, Today at 1 24 02 pm](https://user-images.githubusercontent.com/4511095/138453064-ddb15e1b-1421-4c06-bc7f-48bc2ea0d852.png)

4. Save (persist) the table to the database

The SQL to be executed is:

> CREATE TABLE NewTable (
	ID INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
	CONSTRAINT NewTable_PK PRIMARY KEY (ID)
);

The error message is:

![Prism  Microsoft Remote Desktop, Today at 1 22 32 pm](https://user-images.githubusercontent.com/4511095/138452870-30865f6f-765a-4cd3-9c6a-6f1d7e357c17.png)

**Comment by Sod-Almighty** [comment]
Oddly, if I create the column *without* creating the primary key constraint (i.e. step 3).......DBeaver creates it automatically.

Which is, I suspect, the reason for the error.

**Comment by LonwoLonwo** [comment]
Hello @Sod-Almighty 
It's not a bug, it's a feature.
You can't change autoincrement field in an existing table.

[Documentation](https://www.sqlite.org/autoinc.html) says:
Any attempt to use AUTOINCREMENT on a WITHOUT ROWID table or on a column other than the INTEGER PRIMARY KEY column results in an error.

Therefore, for the convenience of users, we [added](https://github.com/dbeaver/dbeaver/issues/9637) an automatic addition of the Primary key to the autoincrement field when creating a table.

**Comment by Sod-Almighty** [comment]
I realise that it's probably _intended_ to be a feature; but it should gracefully handle the case where the user creates the primary key constraint manually. Either that, or it shouldn't allow the user to do so.
